tools/loader/Loader.cpp

```diff
@@ -427,7 +449,19 @@ Loader::Loader(int argc, char **argv) {
     caffe2NetDescFilename_ = modelPathOpt[0];
     caffe2NetWeightFilename_ = modelPathOpt[1];
   }
+  M_.reset(new Module);
+  std::vector<std::unique_ptr<runtime::DeviceConfig>> configs;
+  if (ExecutionBackend == BackendKind::OpenCL) {
+
+    auto clConfig = new runtime::OpenCLDeviceConfig();
```

Comment:
We need to have some method that accepts the device enum and creates the device config. We should not inline this code into every user (call site).

Reply:
Or may be we could have a virtual method in DeviceManager and ask the DeviceManager to create an empty device config?  This way HM does not need to know the specifics...

Reply:
That would mean we include device discovery and assignment to the runtime. Is that preferred? For example if a user wants to target 3 specific devices out of a pool of 5, how do we let the user of the library select devices? Or is that something we are ok with not supporting?